pkg/controller/proxyconfig/status.go

```diff
@@ -105,8 +107,10 @@ func mergeUserSystemNoProxy(proxy *configv1.Proxy, infra *configv1.Infrastructur
 		set.Insert(clusterNetwork.CIDR)
 	}
 
-	for _, userValue := range strings.Split(proxy.Spec.NoProxy, ",") {
-		set.Insert(userValue)
+	if len(proxy.Spec.NoProxy) != 0 {
+		for _, userValue := range strings.Split(proxy.Spec.NoProxy, ",") {
```

Comment:
nit, but I like [this approach][1] better than your wrapping length guard because it also protects against injecting empty strings for values like `,i.start.with.a.comma.com`.

[1]: https://github.com/openshift/installer/pull/2198/files#diff-ff1d8a3b05383eef08ceaf043a235067R145